src/moment_kinetics.jl

```diff
@@ -416,10 +326,41 @@ function setup_moment_kinetics(input_dict::Dict; restart_prefix_iblock=nothing,
     else
         restarting = true
 
+        if restart === true
+            run_name = input_dict["run_name"]
+            base_directory = get(input_dict, "base_directory", "runs")
+            output_dir = joinpath(base_directory, run_name)
+            io_settings = get(input_dict, "output", Dict{String,Any}())
+            binary_format = get(io_settings, "binary_format", hdf5)
+            if binary_format === hdf5
+                ext = "h5"
+            elseif binary_format === netcdf
+                ext = "cdf"
+            else
+                error("Unrecognized binary_format '$binary_format'")
+            end
+            restart = glob(joinpath(output_dir, run_name * ".dfns*." * ext))[1]
+        end
+
+        # Move the output file being restarted from to make sure it doesn't get
+        # overwritten.
+        dfns_filename, backup_dfns_filename, parallel_io, moments_filename,
+        backup_moments_filename, backup_prefix_iblock =
+            get_backup_filename(restart)
```

Comment:
Would prefer not to use `restart' as a logical and as a filename here.

Reply:
Can agree it's maybe not the clearest to change the value of `restart` that was passed in as an argument. Will make a separate `restart_filename` variable that gets set appropriately.